config/configopaque/maplist.go

```diff
@@ -0,0 +1,116 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package configopaque // import "go.opentelemetry.io/collector/config/configopaque"
+
+import (
+	"cmp"
+	"fmt"
+	"iter"
+	"slices"
+
+	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/confmap/xconfmap"
+)
+
+// Pair is an element of a MapList, and consists of a name and an opaque value.
+type Pair struct {
+	Name  string `mapstructure:"name"`
+	Value String `mapstructure:"value"`
+
+	// prevent unkeyed literal initialization
+	_ struct{}
+}
+
+// MapList is a replacement for map[string]configopaque.String with a similar API,
+// which can also be unmarshalled from (and is stored as) a list of name/value pairs.
+//
+// Pairs are assumed to have distinct names. This is checked during config validation.
```

Comment:
Maybe worth mentioning here that modifying the map through its public API keeps the entries sorted by name

Reply:
Ah, right. Okay, I think it's fine as it is then